nextmove/src/main/java/no/difi/meldingsutveksling/nextmove/ConversationResourceFactory.java

```diff
@@ -0,0 +1,122 @@
+package no.difi.meldingsutveksling.nextmove;
+
+import com.google.common.collect.Lists;
+import lombok.RequiredArgsConstructor;
+import no.difi.meldingsutveksling.config.IntegrasjonspunktProperties;
+import no.difi.meldingsutveksling.nextmove.dpi.DigitalPostInfo;
+import no.difi.meldingsutveksling.nextmove.dpi.FysiskPostInfo;
+import no.difi.meldingsutveksling.nextmove.dpi.PostRetur;
+import no.difi.meldingsutveksling.nextmove.dpi.ReturMottaker;
+import no.difi.meldingsutveksling.serviceregistry.ServiceRegistryLookup;
+import no.difi.meldingsutveksling.serviceregistry.externalmodel.PostAddress;
+import no.difi.meldingsutveksling.serviceregistry.externalmodel.ServiceRecord;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.util.Optional;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static no.difi.meldingsutveksling.ServiceIdentifier.DPI;
+
+@Component
+@RequiredArgsConstructor
+public class ConversationResourceFactory {
+
+    private final ServiceRegistryLookup sr;
+    private final IntegrasjonspunktProperties props;
+
+    public void create(ConversationResource cr) {
+
+        switch (cr.getServiceIdentifier()) {
+            case DPI:
+                setDpiDefaults((DpiConversationResource)cr);
+                break;
+            case DPF:
+            case DPO:
+            case DPV:
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') End this switch case with an unconditional break, return or throw statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS128)